napari/_tests/test_viewer.py

```diff
@@ -302,3 +302,19 @@ class NewLabels(layers.Labels):
     # Make a viewer and add the custom layer
     viewer = make_napari_viewer(show=True)
     viewer.add_layer(NewLabels(np.zeros((10, 10, 10), dtype=np.uint8)))
+
+
+def test_emitting_data_doesnt_change_points_value(make_napari_viewer):
```

Comment:
I think it would be nice to add a similar test that verifies that the cursor position is not changed when the data event is emitted.

Reply:
@kevinyamauchi I've added a test for that now, parameterized for every layer type. Thanks for the suggestion!

Reply:
Looks great! Thank you!